streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroKafkaSpoutTranslator.java

```diff
@@ -104,17 +104,18 @@ public int read () throws IOException {
         }
 
         public int read (byte[] bytes, int off, int len) throws IOException {
-            Preconditions.checkNotNull(bytes, "Given byte array can not be null");
-            Preconditions.checkPositionIndexes(off, len, bytes.length);
+            if (bytes == null) {
+                throw new NullPointerException();
+            } else if (off < 0 || len < 0 || len > bytes.length - off) {
+                throw new IndexOutOfBoundsException();
+            } else if (len == 0) {
```

Comment:
It is good to have this condition to be checked only when ByteBuffer has any bytes to be read. So, It is better to leave where it is currently.

It seems newly added conditional checks follow what `InputStream#(byte b[], int off, int len)` does but the above suggestions are valid for this scenario.